extensions/amp-story/1.0/media-pool.js

```diff
@@ -588,13 +559,12 @@ export class MediaPool {
    *     has been reset.
    */
   resetPoolMediaElementSource_(poolMediaEl) {
-    const mediaType = this.getMediaType_(poolMediaEl);
-    const defaultSources = this.getDefaultSource_(mediaType);
+    const defaultSources = this.getDefaultSource_();
 
     return this.enqueueMediaElementTask_(
       poolMediaEl,
       new UpdateSourcesTask(this.win_, defaultSources)
-    );
+    ).then(() => this.enqueueMediaElementTask_(poolMediaEl, new LoadTask()));
```

Comment:
why is this needed?

Reply:
Gave more context offline, but for the record:

>  I added an extra fix to call load() when removing the sources so it resets the readyState and all we care about.

Reply:
Could you leave a comment to this? This set of functionality could use with some additional documentation.